docs/RPC Spec Generator/index.md

```diff
@@ -0,0 +1,193 @@
+        <element name="BACKGROUND" internal_name="HMI_BACKGROUND" />
+        <element name="NONE" internal_name="HMI_NONE" />
+    </enum>
+```
+
+The HMILevel communicated to the proxies from SDL Core will use the `name` value (`FULL`, `NONE`, etc). However the `HMILevel` enum usage in SDL Core's code will use the `internal_name` value (`HMI_FULL`, `HMI_NONE`, etc).
+
+### struct
+`<struct>` is a complex data type. `<struct>` contains any number of `<param>`. The `<struct>` has a required `"name"` attribute.
+
+Example:
+```xml
+<struct deprecated="boolean" name="string" since="string" until="string">
+  <!--Optional:-->
+  <history>
+    <struct/>
+  </history>
+  <!--Zero or more repetitions:-->
+  <description>string</description>
+  <!--Zero or more repetitions:-->
+  <param array="boolean" defvalue="integer|decimal|boolean|string" deprecated="boolean" mandatory="boolean" maxlength="integer" maxsize="integer" maxvalue="decimal" minlength="integer" minsize="integer" minvalue="decimal" name="string" since="string" type="string" until="string">
+    <!--Zero or more repetitions:-->
+    <description>string</description>
+    <!--Optional:-->
+    <history>
+      <param/>
+    </history>
+  </param>
+</struct>
+```
+
+Each `<param>` requires `"name"`, `"type"` and `"mandatory"` attributes. The `"type"` attribute value should be one of `"Boolean"`, `"Float"`, `"Integer"`, `"String"` or one of the `<enum>`, `<struct>` names specified in the Mobile API.
+
+`<param>` could have `"array"` attribute to represent an array of values or objects of the described type. Attributes `"maxsize"` and `"minsize"` provide additional restrictions to an array.
+
+Numeric types can be restricted using `"minvalue"` and `"maxvalue"`. 
+
+The `"defvalue"` attribute contains default value with different type, depends on `<param>` type, note: this attribute is not allowed for `<struct>` type.
+
+### function
+The `<function>` element represents a specific RPC and message type of the Mobile API. It contains any number of `<param>`. The `<function>` has a required `"name"`, `"messagetype"`, `"functionID"` attributes.
+
+Example:
+```xml
+
+<function deprecated="boolean" functionID="string" messagetype="string" name="string" since="string" until="string">
+  <!--Optional:-->
+  <history>
+    <function/>
+  </history>
+  <!--Zero or more repetitions:-->
+  <description>string</description>
+  <!--Zero or more repetitions:-->
+  <param array="boolean" defvalue="integer|decimal|boolean|string" deprecated="boolean" mandatory="boolean" maxlength="integer" maxsize="integer" maxvalue="decimal" minlength="integer" minsize="integer" minvalue="decimal" name="string" platform="string" since="string" type="string" until="string">
+    <!--Zero or more repetitions:-->
+    <description>string</description>
+    <!--Optional:-->
+    <history>
+      <param/>
+    </history>
+    <!--Zero or more repetitions:-->
+    <todo>string</todo>
```

Comment:
Are these `todo`s expected?

Reply:
Yes the "todo" tag is technically supported. :(